### PR TITLE
Added MKR1000 entry to Boards.h

### PR DESCRIPTION
**Pull request by turkycat**

### CODE REVIEW

File: Boards.h [hunk 1]
```diff
@@ -243,6 +243,23 @@ writePort(port, value, bitmask):  Write an 8 bit port.
...
 
+// Arduino/Genuino MKR1000
+#elif defined(ARDUINO_SAMD_MKR1000)
+#define TOTAL_ANALOG_PINS       7
+#define TOTAL_PINS              22 // 8 digital + 3 spi + 2 i2c + 2 uart + 7 analog
+#define IS_PIN_DIGITAL(p)       (((p) >= 0 && (p) <= 21) && !IS_PIN_SERIAL(p))
+#define IS_PIN_ANALOG(p)        ((p) >= 15 && (p) < 15 + TOTAL_ANALOG_PINS)
+#define IS_PIN_PWM(p)           digitalPinHasPWM(p)
+#define IS_PIN_SERVO(p)         (IS_PIN_DIGITAL(p) && (p) < MAX_SERVOS) // deprecated since v2.4
+#define IS_PIN_I2C(p)           ((p) == 11 || (p) == 12) // SDA = 20, SCL = 21
```
Comment (soundanalogous) [inline Boards.h]: minor nit. The comment is wrong.

File: Boards.h [hunk 2]
```diff
@@ -243,6 +243,23 @@ writePort(port, value, bitmask):  Write an 8 bit port.
...
+#define IS_PIN_SPI(p)           ((p) == SS || (p) == MOSI || (p) == MISO || (p) == SCK) // SS = A2
+#define IS_PIN_SERIAL(p)        ((p) == PIN_SERIAL1_RX || (p) == PIN_SERIAL1_TX)
```
Comment (soundanalogous) [inline Boards.h]: Does the variant file actually define `PIN_SERIAL1_RX` and `PIN_SERIAL1_TX`? Also is SS mapped to A2?

File: Boards.h (hunk 1 above)
Comment (turkycat) [inline Boards.h]: good catch, these changed but the comment didn't.

File: Boards.h (hunk 2 above)
Comment (turkycat) [inline Boards.h]: yes (added comment).
not anymore (removed comment).

**Pull request merged**